core-backend/sandbox/src/env.rs

```diff
@@ -177,9 +176,21 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
 
         log::debug!("execution res = {:?}", res);
 
-        let info: ExtInfo = self.runtime.ext.unset().into_ext_info(|ptr, buff| {
-            self.runtime.memory.read(ptr, buff);
-        });
+        let info = self
+            .runtime
+            .ext
+            .unset()
+            .into_ext_info(|ptr, buff| {
+                self.runtime
+                    .memory
+                    .read(ptr, buff)
+                    .map_err(|_err| "Cannot read sandbox mem")
+            })
+            .map_err(|(reason, gas_amount)| BackendError {
+                reason,
+                description: None,
+                gas_amount,
+            })?;
```

Comment:
Isn't it better to implement `From<(&'static str, u64)> for BackendError` instead of multiple manual conversions?

Reply:
I think we must do this in separate review where we will make refactoring for all errors